src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java

```diff
@@ -194,4 +198,23 @@ public Response deleteDaaForUser(
       return createExceptionResponse(e);
     }
   }
+
+  @POST
+  @PermitAll
+  @Path("/request/{daaId}")
+  public Response sendDaaRequestMessage(
+      @Auth AuthUser authUser,
+      @PathParam("daaId") Integer daaId) {
+    try {
+      User user = userService.findUserByEmail(authUser.getEmail());
+      if (user.getLibraryCards().stream()
+          .anyMatch(libraryCard -> libraryCard.getDaaIds().contains(daaId))) {
+        throw new IllegalArgumentException("User already has this DAA associated with their Library Card");
+      }
+      daaService.sendDaaRequestEmails(user, daaId);
+      return Response.ok().build();
+    } catch (Exception e) {
+      return createExceptionResponse(e);
+    }
```

Comment:
[General Consent question]

Do you know if Consent has a global exception handler, something like [this](https://github.com/DataBiosphere/jade-data-repo/blob/800d4a58827cab52fdcd210260896be9266d0526/src/main/java/bio/terra/app/controller/GlobalExceptionHandler.java)?

Reply:
We don't have a global handler, but all api-level response errors are handled in `createExceptionResponse`: https://broadworkbench.atlassian.net/browse/DUOS-2981

Reply:
@rushtong is that the right link?